numba/cuda/dispatcher.py

```diff
@@ -106,23 +107,50 @@ def __init__(self, py_func, argtypes, link=None, debug=False,
         self._codelibrary = lib
         self.call_helper = cres.call_helper
 
+        # The following are referred to by the cache implementation. Note:
+        # - There are no referenced environments in CUDA.
+        # - Kernels don't have lifted code.
+        # - reload_init is only for parfors.
```

Comment:
Currently it's only used by parfors, but the use extends to anything which needs some sort of "state" reinitialising.